nexmark-flink/src/main/resources/queries-cep/q3.sql

```diff
@@ -0,0 +1,33 @@
+-- -------------------------------------------------------------------------------------------------
+-- Query 3: Longest period of time for average price (Not in original suite)
+-- -------------------------------------------------------------------------------------------------
+-- Finds the longest period of time for which the average price of a bidder did not go below certain threshold.
+-- Illustrates Aggregations in MATCH_RECOGNIZE.
+-- -------------------------------------------------------------------------------------------------
+
+CREATE TABLE discard_sink (
+      bidder  BIGINT,
+      start_tstamp  TIMESTAMP(3),
+      end_tstamp  TIMESTAMP(3),
+      avgPrice  BIGINT
+) WITH (
+    'connector' = 'blackhole'
+);
+
+INSERT INTO discard_sink
+SELECT
+    bidder, start_tstamp, end_tstamp, avgPrice
+FROM bid
+MATCH_RECOGNIZE (
+    PARTITION BY bidder
```

Comment:
I think all the queries should be partitioned by bidder and auction instead of only bidder.